File: src/main/java/com/dh/ondot/member/domain/MemberWithdrawal.java
package com.dh.ondot.member.domain;

import com.dh.ondot.core.BaseTimeEntity;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Table(name = "member_withdrawals")
public class MemberWithdrawal extends BaseTimeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_withdrawal_id")
    private Long id;

    @Column(name = "member_id", nullable = false)
    private Long memberId;

    @Column(name = "withdrawal_reason_id", nullable = false)
    private Long withdrawalReasonId;

    @Column(name = "custom_reason", columnDefinition = "TEXT")
    private String customReason;

    public static MemberWithdrawal create(Long memberId, Long reasonId, String customReason) {
        return MemberWithdrawal.builder()
                .memberId(memberId)
                .withdrawalReasonId(reasonId)
                .customReason(customReason)
                .build();
    }
}

File: src/main/kotlin/com/dh/ondot/member/domain/MemberWithdrawal.kt
package com.dh.ondot.member.domain

import com.dh.ondot.core.BaseTimeEntity
import jakarta.persistence.*

@Entity
@Table(name = "member_withdrawals")
class MemberWithdrawal(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_withdrawal_id")
    val id: Long = 0L,

    @Column(name = "member_id", nullable = false)
    val memberId: Long,

    @Column(name = "withdrawal_reason_id", nullable = false)
    val withdrawalReasonId: Long,

    @Column(name = "custom_reason", columnDefinition = "TEXT")
    val customReason: String? = null,
) : BaseTimeEntity() {

    companion object {
        @JvmStatic
        fun create(memberId: Long, reasonId: Long, customReason: String?): MemberWithdrawal =
            MemberWithdrawal(
                memberId = memberId,
                withdrawalReasonId = reasonId,
                customReason = customReason,
            )
    }
}
